osu.Game.Rulesets.Taiko/Difficulty/Evaluators/ColourEvaluator.cs

```diff
@@ -36,18 +37,64 @@ public static double EvaluateDifficultyOf(RepeatingHitPatterns repeatingHitPatte
             return 2 * (1 - DifficultyCalculationUtils.Logistic(exponent: Math.E * repeatingHitPattern.RepetitionInterval - 2 * Math.E));
         }
 
+        /// <summary>
+        /// Calculates a consistency penalty based on the number of consecutive consistent intervals,
+        /// considering the delta time between each colour sequence.
+        /// </summary>
+        /// <param name="hitObject">The current hitObject to consider.</param>
+        /// <param name="threshold"> The allowable margin of error for determining whether ratios are consistent.</param>
+        private static double consistentRatioPenalty(TaikoDifficultyHitObject hitObject, double threshold = 0.01)
+        {
+            int consistentRatioCount = 0;
+            double totalRatioCount = 0.0;
+
+            TaikoDifficultyHitObject current = hitObject;
+
+            while (current.Previous(1) is TaikoDifficultyHitObject previousHitObject)
+            {
+                double currentRatio = current.Rhythm.Ratio;
+                double previousRatio = previousHitObject.Rhythm.Ratio;
+
+                // If there's no valid hit object before the previous one, break the loop.
+                if (previousHitObject.Previous(1) is not TaikoDifficultyHitObject)
+                    break;
+
+                // A consistent interval is defined as the percentage difference between the two rhythmic ratios with the margin of error.
+                if (Math.Abs(1 - currentRatio / previousRatio) <= threshold)
```

Comment:
Wouldn't it make sense to break when an interval is different or do you want to go through all of the map? 

e: looking at the PR description you indeed want to break on interval change, but I don't see it actually happening?

Reply:
it would. sorted and adjusted to be in line with before